### PR TITLE
Add Absolute Path to Windows and Linux Events

### LINKED ISSUE #768: FSEvents: report absPath as well as relPath in events?
On Windows and Linux events we report the file path relative to the watch root. On Darwin we report the absolute path of the file/dir. We can report both on all platforms.

### PR DESCRIPTION
**Pull request by rnjtranjan**
Fixes #768

### CODE REVIEW

File: src/Streamly/Internal/FileSystem/Event/Linux.hs [hunk 1]
```diff
@@ -949,6 +951,21 @@ getRoot Event{..} =
...
+-- /Internal/
+--
+getAbsPath :: Event -> Array Word8
+getAbsPath ev0 = go ev0
+
+    where
+
+    go ev = unsafePerformIO
+                $ A.fromStream
+                $ (A.toStream (getRoot ev) <> A.toStream (getRelPath ev))
```
Comment (adithyaov) [inline src/Streamly/Internal/FileSystem/Event/Linux.hs:967]: `A.spliceTwo`?

Comment (adithyaov) [inline src/Streamly/Internal/FileSystem/Event/Linux.hs:967]:
```suggestion
    go ev = unsafePerformIO $ A.spliceTwo (getRoot ev) (getRelPath ev)
```

File: src/Streamly/Internal/FileSystem/Event/Windows.hs [hunk 2]
```diff
@@ -483,7 +497,17 @@ getRelPath Event{..} = eventRelPath
...
+
+-- XXX Change the type to Array Word8 to make it compatible with other APIs.
+--
+-- | Get the absolute file system object path for which the event is generated.
+-- The path is a UTF-8 encoded array of bytes.
+--
+-- /Internal/
+--
+getAbsPath :: Event -> String
+getAbsPath ev = getRoot ev <> getRelPath ev
```
Comment (adithyaov) [inline src/Streamly/Internal/FileSystem/Event/Windows.hs:510]: How is this compiling?
```
getRelPath ev :: Array Word8
getRoot ev :: Array Word8
(<>) :: Array Word8 -> Array Word8 -> Array Word8
```

File: src/Streamly/Internal/FileSystem/Event/Linux.hs (hunk 1 above)
Comment (adithyaov) [inline src/Streamly/Internal/FileSystem/Event/Linux.hs:967]:
```suggestion
getAbsPath ev0 = getRoot ev <> getRelPath ev
```

File: src/Streamly/Internal/FileSystem/Event/Windows.hs (hunk 2 above)
Comment (rnjtranjan) [reply src/Streamly/Internal/FileSystem/Event/Windows.hs:510]: in Windows return types are String, 
getRelPath :: Event -> String
getRoot :: Event -> String
getAbsPath :: Event -> String

File: src/Streamly/Internal/FileSystem/Event/Windows.hs [hunk 3]
```diff
@@ -483,7 +497,17 @@ getRelPath Event{..} = eventRelPath
 -- /Internal/
 --
 getRoot :: Event -> String
-getRoot Event{..} = eventRootPath
+getRoot Event{..} = ensureTrailingSlash eventRootPath
+
+-- XXX Change the type to Array Word8 to make it compatible with other APIs.
```
Comment (harendra-kumar) [inline src/Streamly/Internal/FileSystem/Event/Windows.hs:502]: Is there an issue for this?

**Review by harendra-kumar — APPROVED** [review]

File: src/Streamly/Internal/FileSystem/Event/Windows.hs (hunk 3 above)
Comment (harendra-kumar) [reply src/Streamly/Internal/FileSystem/Event/Windows.hs:502]: Please raise an issue if there isn't one already.

**Pull request merged**